jormungandr/src/stats_counter.rs

```diff
@@ -69,4 +73,12 @@ impl StatsCounter {
         }
         .map(SecondsSinceUnixEpoch::from_secs)
     }
+
+    pub async fn set_tip_block(&self, block: Option<Block>) {
+        *self.stats.tip_block.write().await = Arc::new(block);
+    }
+
+    pub async fn get_tip_block(&self) -> Arc<Option<Block>> {
```

Comment:
Try to change this to return `Option<&Block>` and see if it passes the borrow checker at the use site.